cpp/src/arrow/compute/exec/tpch_node_test.cc

```diff
@@ -0,0 +1,709 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
I don't think you need to add this to all the tests but could you add testing of scale factor to one of the tables that depends on it (e.g. partsupp and lineitem)?  I'd suggest testing a smaller scale factor (e.g. 0.25) instead of a larger one to keep test times down.